tests/rpmbuild.at

```diff
@@ -131,6 +131,37 @@ runroot rpmbuild -bs /data/SPECS/specstep.spec 2>&1|grep ^Executing|cut -d: -f1
 [])
 AT_CLEANUP
 
+# ------------------------------
+# Check if rpmbuild -ba *.spec works after 2038
+AT_SETUP([rpmbuild -ba *.spec after 2038])
+AT_KEYWORDS([build])
+RPMDB_INIT
+AT_CHECK([[
+touch -t 204205042355 ${RPMTEST}/data/SPECS/hello.spec
+touch -t 204205042355 ${RPMTEST}/data/SOURCES/hello-1.0.tar.gz
+touch -t 204205042355 ${RPMTEST}/data/SOURCES/hello-1.0-modernize.patch
+
+unset SOURCE_DATE_EPOCH
+run rpmbuild --quiet \
+  -ba  "${RPMTEST}"/data/SPECS/hello.spec
+
+run rpm -qp --qf '[%{FILENAMES} %{FILEMTIMES:date}\n]' "${RPMTEST}"/build/SRPMS/hello-1.0-1.src.rpm
+run rpm -qp --qf '[%{FILENAMES} %{FILEMTIMES:date}\n]' "${RPMTEST}"/build/RPMS/*/hello-1.0-1.*.rpm | grep FAQ
+runroot rpm -i --quiet --nodeps /build/RPMS/*/hello-1.0-1.*.rpm
+run rpm -q --qf '[%{FILENAMES} %{FILEMTIMES:date}\n]' hello-1.0-1 | grep FAQ
+ls -l "${RPMTEST}"/usr/share/doc/hello-1.0/FAQ | cut "-d " -f6-10
```

Comment:
Here too, always work inside the root if at all possible. The paths on the outside are far more difficult to handle and cause all manner of headaches. That's what `runroot_other` is for.